File: js/batch/attack.js
const fs = require("fs");
var arguments = process.argv;
out = arguments[2];
id = arguments[3];
times = arguments[4]
let a = fs.readFileSync(out);
let data = JSON.parse(a);
regex = data["regex"]
attacks = data["attackArrayList"]
attack = attacks[id]
prefix = attack["prefix"]
infix = attack["infix"]
suffix = attack["suffix"]
let attackStr = prefix
for (i = 0; i < times; i++) {
    attackStr = attackStr + infix
}
attackStr = attackStr + suffix

pattern = new RegExp(regex);
let redos = pattern.test(attackStr)

File: src/main/java/cn/ac/ios/Bean/Attack.java
package cn.ac.ios.Bean;

/**
 * @author pqc
 * 这里参数用于其他语言验证的进程通信，不要改动名字和大小写。
 *
 */
public class Attack {

    public String prefix;
    public String infix;
    public String suffix;
    public AttackType type;
    public PatternType patternType;
    public String redos = "";

    public Attack(String prefix, String infix, String suffix, AttackType type, PatternType patternType) {
        this.prefix = prefix;
        this.infix = infix;
        this.suffix = suffix;
        this.type = type;
        this.patternType = patternType;
    }
}
